Tests/NIOHTTP2Tests/ConfiguringPipelineAsyncMultiplexerTests.swift

```diff
@@ -0,0 +1,167 @@
+        let requestCount = 100
+
+
+        let (serverRecorderStream, serverRecorderContinuation) = AsyncStream<InboundRecorder<HTTP2Frame.FramePayload>>.makeStream()
+        var serverRecorderIterator = serverRecorderStream.makeAsyncIterator()
+
+        let clientMultiplexer = try await assertNoThrowWithValue(
+            try await self.clientChannel.configureAsyncHTTP2Pipeline(
+                mode: .client, connectionConfiguration: .init(), streamConfiguration: .init()) { channel -> EventLoopFuture<Channel> in
+                    self.serverChannel.eventLoop.makeSucceededFuture(channel)
+                }.get()
+        )
+
+        let serverMultiplexer = try await assertNoThrowWithValue(
+            try await self.serverChannel.configureAsyncHTTP2Pipeline(
+                mode: .server, connectionConfiguration: .init(), streamConfiguration: .init()) { channel -> EventLoopFuture<Channel> in
+                    let serverRecorder = InboundFramePayloadRecorder()
+                    serverRecorderContinuation.yield(serverRecorder)
+                    return channel.pipeline.addHandlers([OKResponder(), serverRecorder]).map { _ in return channel }
+                }.get()
+        )
+
+        try await assertNoThrow(try await self.assertDoHandshake(client: self.clientChannel, server: self.serverChannel))
+
+
+        let (clientRecorderStream, clientRecorderContinuation) = AsyncStream<InboundRecorder<HTTP2Frame.FramePayload>>.makeStream()
+        var clientRecorderIterator = clientRecorderStream.makeAsyncIterator()
+
+        try await withThrowingTaskGroup(of: Int.self, returning: Void.self) { group in
+            // server
+            group.addTask {
+                var serverInboundChannelCount = 0
+                for try await _ in serverMultiplexer.inbound {
+                    serverInboundChannelCount += 1
+                }
+                return serverInboundChannelCount
+            }
+
+            // client
+            for _ in 0 ..< requestCount {
+                // Let's try sending some requests
+                let streamChannel = try await clientMultiplexer.createStreamChannel { channel -> EventLoopFuture<Channel> in
+                    let clientRecorder = InboundFramePayloadRecorder()
+                    clientRecorderContinuation.yield(clientRecorder)
+                    return channel.pipeline.addHandlers([SimpleRequest(), clientRecorder]).map {
+                        return channel
+                    }
+                }
+
+                try await self.interactInMemory(self.clientChannel, self.serverChannel)
+                try await streamChannel.closeFuture.get()
+
+                let clientRecorder = await clientRecorderIterator.next()!
+                clientRecorder.framesLock.withLock {
+                    clientRecorder.receivedFrames.assertFramePayloadsMatch([ConfiguringPipelineAsyncMultiplexerTests.responseFramePayload])
+                }
```

Comment:
We don't need an async stream here, we can just return the recorder from the init with the channel (or recover it from the channel pipeline) and then check the appropriate frames are there after the stream has been closed.

Note also that you don't need the lock here: `clientRecorder.receviedFrames` acquires the lock for you.